public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -451,6 +451,31 @@ export class LokiDatasource
     return statsForAll;
   }
 
+  async formatQuery(query: string) {
+    const placeHolderScopedVars: any = {
+      __interval: { value: '200y' },
+      __range: { value: '201y' },
+    };
+
+    this.templateSrv.getVariables().forEach((variable) => {
+      if (query.includes(variable.name) === false) {
+        return;
+      }
+
+      placeHolderScopedVars[variable.name] = { value: variable.query };
+    });
+
+    const interpolated = this.interpolateString(query, placeHolderScopedVars);
+    let response = await this.metadataRequest('format_query', { query: interpolated });
+
+    for (const variable in placeHolderScopedVars) {
+      const value = placeHolderScopedVars[variable].value;
+      response = response.replace(new RegExp(`\\[${value}\\]`, 'g'), `[$${variable}]`);
+    }
+
+    return response;
+  }
```

Comment:
For now, it will never be trivial to replace the interpolated value back to the original variable name without the risk of  replacing part of the original query by mistake.

For example, I have this variable called `$var` whose value is `2`:

![2023-03-09 15 29 04](https://user-images.githubusercontent.com/1069378/224056444-202f7478-548c-4436-a526-2406d2969c8f.gif)

Imagine I want to replace after formatting `2` with `$var`, which `2` of the formatted query are we going to replace? We don't know.

I will try to come up with an alternative suggestion _leveraging_ Lezer to replace the values and not using simple string searches.

In the mean time, I would recommend you to remove the interpolation part, and move this code out of the datasource class to avoid adding more complexity to it, like we did in `querySplitting.ts`. It would also make testing easier.